### DISCUSSION TITLE
Stock item management/labels

### DISCUSSION THREAD

**Discussion by wolflu05** [inventree/InvenTree#6475] [Q&A]
I have a lot of drawer shelfs each drawer having its own inventree location. In drawer1 are 10 red LEDs. Now I use 5 of them, and adjust the stock item so that only 5 are remaining. Now I issue a PO with 100 new LEDs. After recieving the PO I check in the parts and it creates a new stock item in drawer1 where I now have 105 red LEDs (stock item 1 (5LEDs) and stock item 2 (100LEDs)).
1. How do I now adjust the stock? Just pick one stock item and remove the taken amount of LEDs? In this case it doesnt really matter what stock item I decrease?
2. What label do I now print on the drawer? 
   - Part labels are not really correct because if I have the LEDs also at a second location I do not know what stock items are in this drawer.
   - Stock item labels are also not correct because if one stock item deplets I have no reference anymore and also I may dont see the other stock items for the same part at the same location directly
   - location labels are also not correct, because I may can have different parts in that location (not likely the case)
   => What about having a new QR code type partlocation that contains both? A part reference and a location reference which shows the stock items for a given part filtered to a location?

Or do you solve that case differently?

**Comment by matmair** [comment]
I use stock locations (and QR codes), items that were delivered together are handled in one stock item wherever possible.
I do not see a general need for an additional label that is a stock location with filters. Seems more confusing than helpful for most users.

**Reply by wolflu05** [reply]
Thanks for your reply, but unfortunately I'm not sure if you understand my problem. You are right with same parts should be one line in a PO so that only one stock item will be created, but what do I do if I order it after a week again? => a new stock item will be created. And in my storage system I don't keep the LEDs from one order separate to the ones from the other order, but instead they are all in one drawer. So I end up with e.g. 30 leds in total, 10 included in stockitem A (PO0001) and 20 in stockitem B (PO0002). And what label should I now print for that drawer? (I already described why each of the 3 already existing labels do not work in my opinion)

And What I tried to explain was not having another label type, but instead having a QR code that contains both that can be generated from any label type that already exists, but I agree that this can add much complexity and would require a lot of dev time, so I'd be happy to be told a different way to solve this.

**Reply by matmair** [reply]
I think you are overcomplicating this. I would have 2 bags (10 pcs, 20 pcs) in the drawer in the described scenario. That drawer would be its own stock location with a stock location QR code printed on the lid.

**Reply by wolflu05** [reply]
Mh, not sure, I currently (without InvenTree for that part) just take out the LEDs from the bag and put them directly in the drawer to the others which are the same model. I need to think a bit deeper about not doing that, because this can get pretty quick a bit unordered, because in the drawers is not that much space for the big individual bags. Am I really the only one doing it like this?

**Reply by SergeoLacruz** [reply]
No you are not :-) 
We also have bulk (Schüttgut) like banana plugs. We use the merge function to combine the stock items. This is OK for parts where
we do not care about the real origin of the component because this gets lost. It cannot destinguised any longer if thas plug was bought
from Mouser or Reichelt.
So we try to keep them apart in most cases because there might be differences that pop up later during production or product usage. 
Michael

**Reply by wolflu05** [reply]
Hello, thanks for that tip with merging stock. That's better than nothing, I'll think about it.
